### ISSUE TITLE
False positive with attributes of type literals

### ISSUE THREAD

**Issue by lijunkang12** [microsoft/pyright#1935]
**Describe the bug**
A bug similar to #1901. The literal information is lost after the definition of a __init__ method.

**To Reproduce**
from typing import Literal
Bar = Literal['bar']

class foo():
    def __init__(self, bar: Bar) -> None:
        self.bar = bar

    def access_bar(self) -> Bar:
        return self.bar
```
Pylance (or Pyright) complains about the `access_bar` method that
```
Expression of type "str" cannot be assigned to return type "Bar"
  "str" cannot be assigned to type "Bar"
```
However, in `__init__`, `self.bar` is correctly typed as `Bar`.

**VS Code extension or command-line**
Pylance 2021.5.4 (with pyright 1.1.144)

**Comment by erictraut** [comment]
This is intended behavior. When inferring the type of an instance variable, literals are "stripped" and the type is widened to its non-literal form. We found that this produced more false positives than the alternative. It's rare that you actually want to limit an instance variable to a literal or a union or literals. If that's the behavior you want, you can provide an explicit type hint in this case:

```python
        self.bar: Bar = bar
```
